src/packages/client/src/runtime/query.ts

```diff
@@ -1369,7 +1369,7 @@ function hasCorrectScalarType(
   // from https://github.com/MikeMcl/decimal.js/blob/master/decimal.js#L115
   if (
     graphQLType === 'String' &&
-    /^(\d+(\.\d*)?|\.\d+)(e[+-]?\d+)?$/i.test(value) &&
+    /^\-?(\d+(\.\d*)?|\.\d+)(e[+-]?\d+)?$/i.test(value) &&
```

Comment:
Would it not be better to have the regex after `expectedType === 'Decimal'` Thus we can skip the regex test if the expected type is not Decimal

Reply:
That is very true!!